packages/lit-dev-content/site/docs/localization/transform-mode.md

```diff
@@ -0,0 +1,285 @@
+
+For example, this `locale-picker` custom element loads a new URL whenever a new
+locale is selected from a drop-down list:
+
+{% switchable-sample %}
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement} from 'lit/decorators.js';
+import {getLocale} from './localization.js';
+import {allLocales} from './generated/locales.js';
+
+@customElement('locale-picker');
+export class LocalePicker extends LitElement {
+  render() {
+    return html`
+      <select @change=${this.localeChanged}>
+        ${allLocales.map(
+          (locale) =>
+            html`<option value=${locale} selected=${locale === getLocale()}>
+              ${locale}
+            </option>`
+        )}
+      </select>
+    `;
+  }
+
+  localeChanged(event: Event) {
+    const newLocale = (event.target as HTMLSelectElement).value;
+    const url = new URL(window.location.href);
+    if (url.searchParams.get('locale') !== newLocale) {
+      url.searchParams.set('locale', newLocale);
+      window.location.assign(url.href);
+    }
+  }
+}
+```
+
+```js
+import {LitElement, html} from 'lit';
+import {getLocale} from './localization.js';
+import {allLocales} from './generated/locales.js';
+
+export class LocalePicker extends LitElement {
+  render() {
+    return html`
+      <select @change=${this.localeChanged}>
+        ${allLocales.map(
+          (locale) =>
+            html`<option value=${locale} selected=${locale === getLocale()}>
+              ${locale}
+            </option>`
+        )}
+      </select>
+    `;
+  }
+
+  localeChanged(event) {
+    const newLocale = event.target.value;
+    const url = new URL(window.location.href);
+    if (url.searchParams.get('locale') !== newLocale) {
+      url.searchParams.set('locale', newLocale);
+      window.location.assign(url.href);
+    }
+  }
+}
+customElements.define('locale-picker', LocalePicker);
+```
+
+{% endswitchable-sample %}
+
+## Rollup
+
+If you use <a href="https://rollupjs.org/" target="_blank"
+rel="noopener">Rollup</a>, and would prefer an integrated solution instead of
+running the `lit-localize build` command separately, import the
+`localeTransformers` function from `@lit/localize-tools/lib/rollup.js` into your
+Rollup config.
+
+This function generates an array of `{locale, transformer}` objects, which you
+can use in conjunction with the
+<a href="https://github.com/rollup/plugins/tree/master/packages/typescript/#transformers" target="_blank" rel="noopener">transformers</a>
+option of
+<a href="https://www.npmjs.com/package/@rollup/plugin-typescript" target="_blank" rel="noopener">@rollup/plugin-typescript</a>
+to generate a separate bundle for each locale.
+
+<div class="alert alert-info">
+
+If you write JavaScript, don't worry about seeing the TypeScript compiler used
+here. Lit Localize depends on the TypeScript compiler to parse, analyze, and
+transform your source code — but it handles plain JavaScript files too!
```

Comment:
No spaces around em dashes... We follow d.g.c. style on this: https://developers.google.com/style/dashes?hl=en

There's a couple more of these, I didn't mark the rest of them.

Reply:
Done. One I turned into a comma.